source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,379 @@ Documents
+           new BasicDBObject("title", "Chronicle of a Death Foretold").append("yearPublished", 1981),
+           new BasicDBObject("title", "Love in the Time of Cholera").append("yearPublished", 1985)));
+
+To insert this document into a collection, instantiate a collection using
+:java-docs:`getCollection()</apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`,
+specifying the ``BasicDBObject`` class. Then, call the
+:doc:`insertOne </usage-examples/insertOne>` operation as follows:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+   MongoCollection<BasicDBObject> collection = database.getCollection("authors", BasicDBObject.class);
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample document
+data from the collection using the following code:
+
+.. code-block:: java
+
+   import com.mongodb.client.model.Filters;
+   // <MongoCollection setup code here>
+
+   BasicDBObject doc = collection.find(Filters.eq("name", "Gabriel García Márquez")).first();
+   if (doc != null) {
+       System.out.println("_id: " + doc.getObjectId("_id")
+           + ", name: " + doc.getString("name")
+           + ", lastUpdated: " + doc.getDate("lastUpdated"));
+
+       BasicDBList novels = (BasicDBList) doc.get("novels");
+       if (novels != null) {
+           BasicDBObject[] novelArr = novels.toArray(new BasicDBObject[0]);
+           for (BasicDBObject novel : novelArr) {
+               System.out.println("title: " + novel.getString("title")
+               + ", yearPublished: " + novel.getInt("yearPublished"));
+           }
+       }
+   }
+
+.. note::
+
+   The code sample above uses helper methods that check the returned type
+   and throw an exception if it is unable to cast the field value.
+   You can call the :java-docs:`get() </apidocs/bson/org/bson/BasicBSONObject.html?is-external=true#get(java.lang.String)>`
+   method to retrieve values as type type ``Object`` and to skip type checking.
+
+The output should resemble the following:
+
+.. code-block:: none
+   :copyable: false
+
+   _id: 5fb5fad05f734e3794741a35, name: Gabriel García Márquez, lastUpdated: Thu Mar 26 00:00:00 EDT 2020
+   title: One Hundred Years of Solitude, yearPublished: 1967
+   title: Chronicle of a Death Foretold, yearPublished: 1981
+   title: Love in the Time of Cholera, yearPublished: 1985
+
+For more information on retrieving and manipulating MongoDB data, see our
+:doc:`CRUD guide </fundamentals/crud>`.
+
+For more information on this class, see the
+:java-docs:`BasicDBObject API documentation </apidocs/mongodb-driver-core/com/mongodb/BasicDBObject.html>`.
+
+Summary
+-------
+
+In this guide, we covered the following topics on Java document classes:
```

Comment:
`Java document classes` is kind of awkward.  It's Java driver classes, not Java classes.  And they're not document classes, they're classes used to model/represent BSON documents.

Can you rephrase?

Reply:
Sure, found a couple other similar mentions in this PR, will update.